src/js/relayserver/RelayServer.js

```diff
@@ -0,0 +1,558 @@
+    debug('gasLimit', gas)
+    const nonce = await this._getNonce()
+    debug('nonce', nonce)
+    // TODO: change to eip155 chainID
+    const txToSign = new Transaction({
+      from: this.address,
+      to: destination,
+      value: value || 0,
+      gas,
+      gasPrice,
+      data: encodedCall ? Buffer.from(encodedCall.slice(2), 'hex') : Buffer.alloc(0),
+      nonce
+    })
+    spam('txToSign', txToSign)
+    const signedTx = this.keyManager.signTransaction(txToSign)
+    const storedTx = new StoredTx({
+      from: txToSign.from,
+      to: txToSign.to,
+      value: txToSign.value,
+      gas: txToSign.gas,
+      gasPrice: txToSign.gasPrice,
+      data: txToSign.data,
+      nonce: txToSign.nonce,
+      txId: ethUtils.bufferToHex(txToSign.hash()),
+      attempts: 1
+    })
+    await this.txStoreManager.putTx({ tx: storedTx })
+    this.nonce++
+    const receipt = await this.web3.eth.sendSignedTransaction(signedTx)
+    console.log('\ntxhash is', receipt.transactionHash)
+    if (receipt.transactionHash.toLowerCase() !== storedTx.txId.toLowerCase()) {
+      throw new Error(`txhash mismatch: from receipt: ${receipt.transactionHash} from txstore:${storedTx.txId}`)
+    }
+    return { receipt, signedTx }
+  }
+
+  async _resendTransaction ({ tx }) {
+    // Calculate new gas price as a % increase over the previous one
+    let newGasPrice = parseInt(tx.gasPrice * retryGasPriceFactor)
+    // Sanity check to ensure we are not burning all our balance in gas fees
+    if (newGasPrice > maxGasPrice) {
+      console.log('Capping gas price to max value of', maxGasPrice)
+      newGasPrice = maxGasPrice
+    }
+    // Resend transaction with exactly the same values except for gas price
+    const txToSign = new Transaction({
+      from: tx.from,
+      to: tx.to,
+      value: tx.value,
+      gas: tx.gas,
+      gasPrice: newGasPrice,
+      data: tx.data,
+      nonce: tx.nonce
+    })
+    spam('txToSign', txToSign)
+    // TODO: change to eip155 chainID
+    const signedTx = this.keyManager.signTransaction(txToSign)
+    const storedTx = new StoredTx({
+      from: txToSign.from,
+      to: txToSign.to,
+      value: txToSign.value,
+      gas: txToSign.gas,
+      gasPrice: txToSign.gasPrice,
+      data: txToSign.data,
+      nonce: txToSign.nonce,
+      txId: ethUtils.bufferToHex(txToSign.hash()),
+      attempts: tx.attempts + 1
+    })
+    await this.txStoreManager.putTx({ tx: storedTx })
+    debug('resending tx with nonce', txToSign.nonce)
+    debug('account nonce', await this.web3.eth.getTransactionCount(this.address))
+    const receipt = await this.web3.eth.sendSignedTransaction(signedTx)
+    console.log('\ntxhash is', receipt.transactionHash)
+    if (receipt.transactionHash.toLowerCase() !== storedTx.txId.toLowerCase()) {
+      throw new Error(`txhash mismatch: from receipt: ${receipt.transactionHash} from txstore:${storedTx.txId}`)
+    }
+    return { receipt, signedTx }
+  }
+
+  async _getNonce () {
```

Comment:
It is keeping track of its nonce. Notice that the check here is if only for some reason the relay nonce is behind the blockchain one, not vice versa. The whole point of keeping track of the nonce locally is to avoid signing different transactions with the same nonce, and this cannot happen here as we only advance the nonce forward.

Reply:
It was called pollNonce in Go, I can rename.